pkg/settlement/swap/swap.go

```diff
@@ -125,6 +125,12 @@ func (s *Service) Pay(ctx context.Context, peer swarm.Address, amount uint64) er
 	if err != nil {
 		return err
 	}
+
+	balance, err := s.chequebook.AvailableBalance(ctx)
```

Comment:
a peer's accounting lock is held during `Pay`. we should avoid adding additional blockchain calls here if possible.

Reply:
I'd like to ideally avoid the complexities of managing more goroutines. Would it be an idea to add the current available balance as a return value on the `unreserveTotalIssued` and then add it to the return value of `Issue`?

Reply:
the call is in `reserve` not `unreserve`. it's also not the correct value but you should get the right one if you reduce it by `amount`.

Reply:
whoops sorry my mistake. so adding the return value to `reserve`?

Reply:
yeah. actually it might also be fine (for now at least) to have this the only point where it is updated as this is the only place where low available balance actual matters (other than a withdraw call) and this will be called quite frequently so the metric should not get too old.